storage/src/main/java/org/apache/kafka/storage/internals/log/AsyncOffsetReader.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.storage.internals.log;
+
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.record.FileRecords;
+import org.apache.kafka.storage.internals.epoch.LeaderEpochFileCache;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+/**
+ * Interface used to decouple UnifiedLog and RemoteLogManager.
+ */
+public interface AsyncOffsetReader {
+
+    /**
+     * Retrieve the offset for the specified timestamp. UnifiedLog may call this method when handling OffsetRequest > 0
```

Comment:
> when handling OffsetRequest > 0

I couldn't understand, What does it mean?

Reply:
It means "handling ListOffsets requests > version 0". I pushed an update rewording the comments

Reply:
still not clear on this comment. We support asyncRemoteListOffsets from v10: 

https://sourcegraph.com/github.com/apache/kafka/-/blob/clients/src/main/resources/common/message/ListOffsetsRequest.json?L42

Reply:
I've pushed an update removing the comments about ListOffsets v0 as this version was removed in Kafka 4.0.

Reply:
ack.